src/main/java/htsjdk/samtools/StreamInflatingIndexingOutputStream.java

```diff
@@ -26,7 +27,7 @@ public StreamInflatingIndexingOutputStream(final OutputStream s1, final File ind
             this.s1 = s1;
             this.s2 = new PipedOutputStream();
             final PipedInputStream pin = new PipedInputStream(this.s2, Defaults.NON_ZERO_BUFFER_SIZE);
-            this.thread = new Thread(new Indexer(indexFile, pin), "BamIndexingThread");
+            this.thread = new Thread(new Indexer(indexFile.toPath(), pin), "BamIndexingThread");
```

Comment:
Could it be possible to add a constructor for `Path` also here, and delegate the `File`one to that one?